modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/CommonBeanBaseDefinitionCreator.java

```diff
@@ -0,0 +1,152 @@
+    BeanDefinitionBuilder beanDefinitionBuilder =
+        createBeanDefinitionBuilder(request.getSpringComponentModel(), buildingDefinition);
+    processAnnotations(request.getSpringComponentModel(), beanDefinitionBuilder);
+    processComponentDefinitionModel(springComponentModels, request, buildingDefinition, beanDefinitionBuilder);
+
+    return true;
+  }
+
+  private BeanDefinitionBuilder createBeanDefinitionBuilder(SpringComponentModel componentModel,
+                                                            ComponentBuildingDefinition buildingDefinition) {
+    BeanDefinitionBuilder beanDefinitionBuilder;
+    if (buildingDefinition.getObjectFactoryType() != null) {
+      beanDefinitionBuilder = createBeanDefinitionBuilderFromObjectFactory(componentModel, buildingDefinition);
+    } else {
+      beanDefinitionBuilder = genericBeanDefinition(componentModel.getType());
+    }
+    return beanDefinitionBuilder;
+  }
+
+  protected final void processNestedAnnotations(ComponentAst component, Map<QName, Object> previousAnnotations) {
+    if (component == null) {
+      return;
+    }
+    component.directChildrenStream()
+        .filter(cdm -> cdm.getIdentifier().equals(ANNOTATIONS_ELEMENT_IDENTIFIER))
+        .findFirst()
+        .ifPresent(annotationsCdm -> annotationsCdm.directChildrenStream()
+            .forEach(annotationCdm -> previousAnnotations
+                .put(new QName(annotationCdm.getIdentifier().getNamespaceUri(),
+                               annotationCdm.getIdentifier().getName()),
+                     annotationCdm.getRawParameterValue(BODY_RAW_PARAM_NAME).orElse(null))));
+  }
+
+  protected final void processAnnotations(SpringComponentModel componentModel, BeanDefinitionBuilder beanDefinitionBuilder) {
+    if (Component.class.isAssignableFrom(componentModel.getType())
+        // ValueResolver end up generating pojos from the extension whose class is enhanced to have annotations
+        || ValueResolver.class.isAssignableFrom(componentModel.getType())) {
+      Map<QName, Object> annotations =
+          processMetadataAnnotationsHelper(beanDefinitionBuilder, componentModel.getComponent());
+      processNestedAnnotations(componentModel.getComponent(), annotations);
+      if (!annotations.isEmpty()) {
+        beanDefinitionBuilder.addPropertyValue(ANNOTATIONS_PROPERTY_NAME, annotations);
+      }
+    }
+  }
+
+  private Map<QName, Object> processMetadataAnnotationsHelper(BeanDefinitionBuilder builder, ComponentAst componentModel) {
+    Map<QName, Object> annotations = new HashMap<>();
+    if (componentModel == null) {
+      return annotations;
+    } else {
+      if (Component.class.isAssignableFrom(builder.getBeanDefinition().getBeanClass())) {
+        addMetadataAnnotationsFromDocAttributes(annotations, componentModel.getMetadata());
+        builder.getBeanDefinition().getPropertyValues().addPropertyValue(ANNOTATIONS_PROPERTY_NAME, annotations);
+      }
+
+      return annotations;
+    }
```

Comment:
You might be able to do this in a single if. Two of the paths this current code takes end up returning a new HashMap